packages/cw721/src/execute.rs

```diff
@@ -0,0 +1,810 @@
+pub fn migrate_legacy_minter_and_creator(
+    storage: &mut dyn Storage,
+    api: &dyn Api,
+    _env: &Env,
+    _msg: &Cw721MigrateMsg,
+    response: Response,
+) -> Result<Response, Cw721ContractError> {
+    let minter = MINTER.item.may_load(storage)?;
+    // no migration in case minter is already set
+    if minter.is_some() {
+        return Ok(response);
+    }
+    // in v0.17/18 cw_ownable::OWNERSHIP was used for minter, now it is used for creator
+    let ownership_previously_used_as_minter = CREATOR.item.may_load(storage)?;
+    let creator_and_minter = match ownership_previously_used_as_minter {
+        // v0.18 migration
+        Some(ownership) => {
+            // owner is used for both: creator and minter
+            // since it is already set for creator, we only need to migrate minter
+            let owner = ownership.owner.map(|a| a.to_string());
+            MINTER.initialize_owner(storage, api, owner.as_deref())?;
+            owner
+        }
+        // v0.17 and older migration
+        None => {
+            let legacy_minter_store: Item<Addr> = Item::new("minter");
+            let legacy_minter = legacy_minter_store.load(storage)?;
+            MINTER.initialize_owner(storage, api, Some(legacy_minter.as_str()))?;
+            CREATOR.initialize_owner(storage, api, Some(legacy_minter.as_str()))?;
+            Some(legacy_minter.to_string())
+        }
+    };
+    Ok(response.add_attribute("creator_and_minter", none_or(creator_and_minter.as_ref())))
+}
+
+/// Migrates only in case collection_info is not present
+pub fn migrate_legacy_collection_info(
+    storage: &mut dyn Storage,
+    env: &Env,
+    _msg: &Cw721MigrateMsg,
+    response: Response,
+) -> Result<Response, Cw721ContractError> {
+    let contract = Cw721Config::<
+        DefaultOptionMetadataExtension,
+        Empty,
+        Empty,
+        Empty,
+        DefaultOptionCollectionInfoExtension,
+    >::default();
+    match contract.collection_info.may_load(storage)? {
+        Some(_) => Ok(response),
+        None => {
+            // contract info is legacy collection info
+            let legacy_collection_info_store: Item<cw721_016::ContractInfoResponse> =
+                Item::new("nft_info");
+            let legacy_collection_info = legacy_collection_info_store.load(storage)?;
+            let collection_info = CollectionInfo {
+                name: legacy_collection_info.name.clone(),
+                symbol: legacy_collection_info.symbol.clone(),
+                extension: None,
+                updated_at: env.block.time,
+            };
+            contract.collection_info.save(storage, &collection_info)?;
+            Ok(response
+                .add_attribute("migrated collection name", legacy_collection_info.name)
+                .add_attribute("migrated collection symbol", legacy_collection_info.symbol))
+        }
+    }
```

Comment:
... if None, then copy old contract info to new collection info. Please note here, there are new props:
- `extension` for collection info extension
- `updated_at` as timestamp, this allows e.g. for frontend and ics721 checking whether collection data has changed!